aten/src/TH/generic/THTensor.cpp

```diff
@@ -174,41 +159,25 @@ THTensor *THTensor_(newContiguous)(THTensor *self)
 
 THTensor *THTensor_(newSelect)(THTensor *tensor, int dimension_, int64_t sliceIndex_)
 {
-  THTensor *self = THTensor_(newWithTensor)(tensor);
+  THTensor *self = at::native::alias(THTensor_wrap(tensor)).unsafeGetTensorImpl();
```

Comment:
nit: I think it's cleaner to just leave `THTensor_(newWithTensor)` and have it do this internally.

When there's only 1 call like with newView I think the tradeoff is in the other direction.

Reply:
don't we want to remove `THTensor_(newWithTensor)`?

Reply:
we definitely want to remove the binding and the separate implementation.

We want to remove the method as well, but I don't think the aesthetics are worth it at this point: you'd have to move most of the callers to make it not ugly IMO.